### ISSUE TITLE
[FEAT] ... public_network_access_enabled option not part of deployment

### ISSUE THREAD

**Issue by leonavevor** [claranet/terraform-azurerm-db-postgresql-flexible#5]
### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

currently there is no support this "public_network_access_enabled = false"
would be nice have in place.

### New or Affected Resource(s)/Data Source(s)

none

### Potential Terraform Configuration

```hcl
module "postgresql_flexible" {
  source  = "claranet/db-postgresql-flexible/azurerm"
  version = "7.3.0"

  name_prefix         = "postgresql-flexible-server"
  ...
  public_network_access_enabled = false
  ...
}
```

### References

https://registry.terraform.io/modules/Azure/postgresql/azurerm/latest

**Comment by Shr3ps** [comment]
Hello,

thanks for your issue, but you're linking https://registry.terraform.io/modules/Azure/postgresql/azurerm/latest which is deploying a deprecated Azure PostgreSQL instead of Azure PostgreSQL Flexible.

But I dig a little, and found this issue on the official provider for the Flexible resource: https://github.com/hashicorp/terraform-provider-azurerm/issues/24708

**Comment by Shr3ps** [comment]
Fixed with module v7.4.0 release
